samza-core/src/main/java/org/apache/samza/zk/ZkUtils.java

```diff
@@ -84,8 +86,50 @@ public static ZkConnection createZkConnection(String zkConnectString, int sessio
     return new ZkConnection(zkConnectString, sessionTimeoutMs);
   }
 
-  public static ZkClient createZkClient(ZkConnection zkConnection, int connectionTimeoutMs) {
-    return new ZkClient(zkConnection, connectionTimeoutMs);
+  /**
+   * helper method to create zkClient
+   * @param connectString - zkConnect string
+   * @param sessionTimeoutMS - session timeout
+   * @param connectionTimeoutMs - connection timeout
+   * @return zkClient object
+   */
+  public static ZkClient createZkClient(String connectString, int sessionTimeoutMS, int connectionTimeoutMs) {
```

Comment:
Not sure how much can be done within the scope of this PR. I'm happy to do the refactoring as a part of a later PR, and submit one.

1. Can we move the `initPath` inside the `ZkUtils`? That way, the lifecycle of initializing `ZkClient` is not spread across multiple places? Optionally, `ZkUtils` can take in a `ZkConfig` object in its constructor. 

2. Also, can we get rid off two ways (currently exposed as 2 factory methods) to obtain an instance of `ZkClient` where in one case you obtained a pre-inited object, and in the other you don't?

3. Also, it's slightly strange to have the name `ZkUtils` , Usually `Util` classes are a bunch of static methods, with a private constructor. Instead it should be something like `ZkClientWrapper` or `SamzaZkClient` that is a thin wrapper around `ZkClient` with well-defined responsibilities.